esp/bindings/http/platform/httpservice.cpp

```diff
@@ -974,3 +875,552 @@ int CEspHttpServer::onGet()
+    const char* userName = (authReq.requestParams) ? authReq.requestParams->queryProp("username") : NULL;
+    const char* password = (authReq.requestParams) ? authReq.requestParams->queryProp("password") : NULL;
+    if (!isEmptyString(userName) && !isEmptyString(password))
+        return doSessionAuth(createHTTPSession(authReq, nullptr, "/"), authReq, userName, password);
+
+    if(authReq.isSoapPost) //from SOAP Test page
+        sendMessage("Session expired. Please close this page and login again.", "text/html; charset=UTF-8");
+    else //from other page
+        askUserLogin(authReq.authBinding, createHTTPSession(authReq, nullptr, nullptr));
+    return authFailed;
+}
+
+EspAuthState CEspHttpServer::checkUserAuthPerRequest(EspAuthRequest& authReq)
+{
+    ESPLOG(LogMax, "checkUserAuthPerRequest");
+
+    authReq.authBinding->populateRequest(m_request.get());
+    if (authReq.authBinding->doAuth(authReq.ctx))
+    {//We do pass the authentication per the request
+        // authenticate optional groups. Do we still need?
+        authOptionalGroups(authReq);
+
+        StringBuffer userName, peer;
+        ESPLOG(LogNormal, "Authenticated for %s@%s", authReq.ctx->getUserID(userName).str(), m_request->getPeer(peer).str());
+        return authSucceeded;
+    }
+    if(!authReq.isSoapPost)
+        return authUnknown;
+
+    //If SoapPost, username/password may be in soap:Header which is not in HTTP header.
+    //The doAuth() may check them inside CSoapService::processHeader() later.
+    authReq.ctx->setToBeAuthenticated(true);
+    return authPending;
+}
+
+void CEspHttpServer::sendMessage(const char* msg, const char* msgType)
+{
+    m_response->setContent(msg);
+    m_response->setContentType(msgType);
+    m_response->setStatus(HTTP_STATUS_OK);
+    m_response->send();
+}
+
+EspAuthState CEspHttpServer::doSessionAuth(unsigned sessionID, EspAuthRequest& authReq, const char* _userName, const char* _password)
+{
+    ESPLOG(LogMax, "doSessionAuth: %s<%u>", PropSessionID, sessionID);
+
+    CDateTime now;
+    now.setNow();
+    time_t timeNow = now.getSimple();
+    Owned<IRemoteConnection> conn = querySDSConnection(authReq.authBinding->queryDomainSessionSDSPath(), RTM_LOCK_WRITE, SESSION_SDS_LOCK_TIMEOUT);
+    IPropertyTree* domainSessions = readAndCleanDomainSessions(authReq.authBinding, conn, timeNow);
+    VStringBuffer xpath("%s[%s='%u']", PathSessionSession, PropSessionID, sessionID);
+    IPropertyTree* sessionTree = domainSessions->queryBranch(xpath.str());
+    if (!sessionTree)
+    {
+        ESPLOG(LogMin, "Authentication failed: session:<%u> not found", sessionID);
+        if(authReq.isSoapPost) //from SOAP Test page
+            sendMessage("Session expired. Please close this page and login again.", "text/html; charset=UTF-8");
+        else
+            askUserLogin(authReq.authBinding, createHTTPSession(authReq, domainSessions, nullptr));
+
+        commitAndCloseRemoteConnection(conn);
```

Comment:
See comment near implementation, I don't see why calling this method is necessary.
(and true of most if not all of other calls to this method)